File: problems/2496-maximum-value-of-a-string-in-an-array/solution.java
class Solution {
    public int maximumValue(String[] strs) {
        int maxValue = 0;
        for (String str : strs) {
            boolean isNumeric = true;
            for (int i = 0; i < str.length(); i++) {
                if (!Character.isDigit(str.charAt(i))) {
                    isNumeric = false;
                    break;
                }
            }
            int value;
            if (isNumeric) {
                value = Integer.parseInt(str);
            } else {
                value = str.length();
            }
            maxValue = Math.max(maxValue, value);
        }
        return maxValue;
    }
}

File: problems/2496-maximum-value-of-a-string-in-an-array/solution.py
class Solution:
    def maximumValue(self, strs: List[str]) -> int:
        max_value = 0
        for s in strs:
            if s.isdigit():
                max_value = max(max_value, int(s))
            else:
                max_value = max(max_value, len(s))
        return max_value